hubblestack/extmods/returners/splunk_osqueryd_return.py

```diff
@@ -40,176 +40,180 @@
+    """
+    Get osqueryd data and post it to Splunk
+    """
+    data = ret['return']
+    if not data:
+        return
+    host_args = _build_args(ret)
+    # Get cloud details
+    cloud_details = __grains__.get('cloud_details', {})
 
+    try:
+        opts_list = get_splunk_options(sourcetype='hubble_osqueryd',
+                                       add_query_to_sourcetype=True,
+                                       _nick={'sourcetype_osqueryd': 'sourcetype'})
         for opts in opts_list:
-            logging.debug('Options: %s' % json.dumps(opts))
-            custom_fields = opts['custom_fields']
-
-            # Set up the fields to be extracted at index time. The field values must be strings.
-            # Note that these fields will also still be available in the event data
-            index_extracted_fields = []
-            try:
-                index_extracted_fields.extend(__opts__.get('splunk_index_extracted_fields', []))
-            except TypeError:
-                pass
-
+            logging.debug('Options: %s', json.dumps(opts))
             # Set up the collector
             args, kwargs = make_hec_args(opts)
             hec = http_event_collector(*args, **kwargs)
-
-            data = ret['return']
-            minion_id = ret['id']
-            jid = ret['jid']
-            fqdn = __grains__['fqdn']
-            # Sometimes fqdn is blank. If it is, replace it with minion_id
-            fqdn = fqdn if fqdn else minion_id
-            try:
-                fqdn_ip4 = __grains__.get('local_ip4')
-                if not fqdn_ip4:
-                    fqdn_ip4 = __grains__['fqdn_ip4'][0]
-            except IndexError:
-                try:
-                    fqdn_ip4 = __grains__['ipv4'][0]
-                except IndexError:
-                    raise Exception('No ipv4 grains found. Is net-tools installed?')
-            if fqdn_ip4.startswith('127.'):
-                for ip4_addr in __grains__['ipv4']:
-                    if ip4_addr and not ip4_addr.startswith('127.'):
-                        fqdn_ip4 = ip4_addr
-                        break
-            local_fqdn = __grains__.get('local_fqdn', __grains__['fqdn'])
-
-            # Sometimes fqdn reports a value of localhost. If that happens, try another method.
-            bad_fqdns = ['localhost', 'localhost.localdomain', 'localhost6.localdomain6']
-            if fqdn in bad_fqdns:
-                new_fqdn = socket.gethostname()
-                if '.' not in new_fqdn or new_fqdn in bad_fqdns:
-                    new_fqdn = fqdn_ip4
-                fqdn = new_fqdn
-
-            # Get cloud details
-            cloud_details = __grains__.get('cloud_details', {})
-
-            if not data:
-                return
-            else:
-                for query_results in data:
-                    event = {}
-                    query_name = query_results['name']
-                    event.update({'query': query_name})
```

Comment:
I wonder if we could use `hubblestack.utils.stdrec.update_payload()` here? If it can't do the this section, perhaps it could be taught to do so? Then the 4 or 6 PRs that all seem to have this section could be using the exact same setup.

Though, I think the non-splunk returners might need somewhat different time fields.